pnpm-lock.yaml

```diff
@@ -20,12 +20,16 @@ specifiers:
   eslint-plugin-n: ^15.6.1
   eslint-plugin-prefer-let: ^3.0.1
   eslint-plugin-promise: ^6.1.1
+  hotscript: ^1.0.8
   nanodelay: ^2.0.2
   size-limit: ^8.2.4
   tsm: ^2.3.0
   typescript: ^4.9.5
   uvu: ^0.5.6
 
+dependencies:
+  hotscript: 1.0.8
```

Comment:
Can we avoid dependency (for insurance by copying types helpers)?

Right now it is zero-dependency project which is one of the argument why it is minimalistic. I am ok to loose this feature for a big feature, but not here.

Reply:
> What kind of insurance?

Can we avoid dependency (for *instance by copying types helpers)?

Reply:
I mean, insurance from what exactly?

Reply:
No insurance, it was just typo.

Having extra dependency is loosing one of our feature. How many code do we use from dependency? Can we just copy it?

Reply:
We use quite a lot of their code. These are just two type functions (`Objects.AllPaths` and `GetFromPath`), but those are tightly entangled with the rest of the lib, and for a good reason (making the same features without this entanglement is quite hard, I was talking about this in Slack).
If I were to copy everything those two function rely on I would copy at least 150-200 LOCs of code that I couldn't support in any way. And I can't really say if I can write those helpers myself in a meaningful time.

Still, I don't understand what you mean by "zero dependency". This is pure TS types, it won't bloat the bundle in any way. I would think "zero dependency" is mostly about code-emitting deps. We can rename the feature from "Zero dependencies" to "Zero runtime dependencies" 😂

Reply:
On top of that, if we one day decide to move away from `.js` + `.d.ts` to plain `.ts`, `hotscript` would actually simplify most of the typings we have.

Reply:
> This is pure TS types, it won't bloat the bundle in any way

People will not try to understand what type of dependency do they see. They still will see a dependency and will think about the project that it is not so `nano-` anymore.